src/devices/tuya.ts

```diff
@@ -6099,6 +6099,78 @@ const definitions: Definition[] = [
+                    .withSystemMode(['cool', 'heat', 'fan_only'])
+                    .withFanMode(['low', 'medium', 'high', 'auto'], ea.STATE_SET)
+                    .withLocalTemperatureCalibration(-5, 5, 0.5, ea.STATE_SET),
+            e.min_temperature().withValueMin(5).withValueMax(15),
+            e.max_temperature().withValueMin(15).withValueMax(45),
+            exposes.switch().withState('eco_mode', false, 'ECO Mode OFF/ON'),
+            e.max_temperature_limit().withDescription('ECO Heating energy-saving temperature (Default: 20 ºC)').withValueMin(15).withValueMax(30),
+            e.min_temperature_limit().withDescription('ECO Cooling energy-saving temperature (Default: 26 ºC)').withValueMin(15).withValueMax(30),
+            e.deadzone_temperature().withValueMin(0).withValueMax(5).withValueStep(1),
+            e.binary('valve', ea.STATE, 'CLOSED', 'OPEN').withDescription('3-Way Valve status'),
+        ],
+        meta: {
+            tuyaDatapoints:
+            [
+                [1, 'state', tuya.valueConverter.onOff],
+                [2, 'system_mode', tuya.valueConverterBasic.lookup({'cool': tuya.enum(0), 'heat': tuya.enum(1), 'fan_only': tuya.enum(2)})],
+                [4, 'eco_mode', tuya.valueConverter.onOff],
+                [16, 'current_heating_setpoint', tuya.valueConverter.divideBy10],
+                [19, 'max_temperature', tuya.valueConverter.divideBy10],
+                [24, 'local_temperature', tuya.valueConverter.divideBy10],
+                [26, 'min_temperature', tuya.valueConverter.divideBy10],
+                [27,  'local_temperature_calibration',  {
+                    from: (value) => {
+                        if (value > 4000) 
+                            return value - 4096;
+                        return value;
+                        },
+                    to: (value) => {
+                        if (value < 0) 
+                            return 4096 + value;
+                        return value;
+                        }
+                    }],
+                [28, 'fan_mode', tuya.valueConverterBasic.lookup({'low': tuya.enum(0), 'medium': tuya.enum(1), 'high': tuya.enum(2), 'auto': tuya.enum(3)})],
+                [36, 'valve', tuya.valueConverterBasic.lookup({'OPEN': 0, 'CLOSE': 1})],
+                [40, 'child_lock', tuya.valueConverter.lockUnlock],
+                [103, 'deadzone_temperature', tuya.valueConverter.raw],
+                [104, 'min_temperature_limit', tuya.valueConverter.divideBy10],
+                [105, 'max_temperature_limit', tuya.valueConverter.divideBy10],
+                /* TODO - help needed
+                "101": DP for "Schedule/Manual",
+                "106": DP for "Schedule",
+                "107": DP for "Schedule",
+                "108": DP for "Schedule",
+                "109": DP for "Schedule",
+                "110": DP for "Schedule",
+                "111": DP for "Schedule",
+                "112": DP for "Schedule"
```

Comment:
Can you check the schedule for other tuya thermostats and see if that works? 
- https://github.com/Koenkk/zigbee-herdsman-converters/blob/1f73585e16543af694ef742033ef9bb63ef18594/src/devices/tuya.ts#L2623
- https://github.com/Koenkk/zigbee-herdsman-converters/blob/1f73585e16543af694ef742033ef9bb63ef18594/src/devices/tuya.ts#L2660

Reply:
Hi, Koenkk.
Added this:
```javascript
toZigbee: [{...tuya.tz.datapoints, key: [...tuya.tz.datapoints.key, 'fan_mode', 'deadzone_temperature', 'eco_mode', 'max_temperature_limit', 'min_temperature_limit', 'manual_mode']}],
``` 
and:
```javascript
    	e.binary('manual_mode', ea.STATE_SET, 'ON', 'OFF').withDescription('Manual = ON or Schedule = OFF'),
    	...tuya.exposes.scheduleAllDays(ea.STATE_SET, 'HH:MM/C HH:MM/C HH:MM/C HH:MM/C HH:MM/C HH:MM/C'),
``` 
Finally:
```javascript
			[106, 'schedule_sunday', tuya.valueConverter.thermostatScheduleDayMultiDPWithDayNumber(7)],
			[107, 'schedule_saturday', tuya.valueConverter.thermostatScheduleDayMultiDPWithDayNumber(6)],
			[108, 'schedule_friday', tuya.valueConverter.thermostatScheduleDayMultiDPWithDayNumber(5)],
			[109, 'schedule_thursday', tuya.valueConverter.thermostatScheduleDayMultiDPWithDayNumber(4)],
			[110, 'schedule_wednesday', tuya.valueConverter.thermostatScheduleDayMultiDPWithDayNumber(3)],
			[111, 'schedule_tuesday', tuya.valueConverter.thermostatScheduleDayMultiDPWithDayNumber(2)],
			[112, 'schedule_monday', tuya.valueConverter.thermostatScheduleDayMultiDPWithDayNumber(1)],
			[101, 'manual_mode', tuya.valueConverter.onOff],
``` 

This is working flawlessly, although I think there is a better/fancy way to do it.
Thank you for your time.

Reply:
Sorry, but there's something that must be changed also:
```javascript
onEvent: tuya.onEventSetLocalTime,
``` 
instead of:
```javascript
onEvent: tuya.onEventSetTime,
``` 
Without this mod, the days of the week are out of sync, for example, today is Monday but it synchronizes as Saturday.

Reply:
Great, could you update your PR?

Reply:
I've tried but it fails and I don't know how to solve it:
```javascript
Error: src/devices/tuya.ts(6155,17): error TS2322: Type '(value: unknown) => unknown' is not assignable to type '(value: unknown, meta?: Meta, options?: KeyValue, publish?: Publish) => string | number | boolean | KeyValue'.
  Type 'unknown' is not assignable to type 'string | number | boolean | KeyValue'.
Error: src/devices/tuya.ts(6156,25): error TS2365: Operator '>' cannot be applied to types 'unknown' and 'number'.
Error: src/devices/tuya.ts(6[15](https://github.com/Koenkk/zigbee-herdsman-converters/actions/runs/6239136324/job/16936417150?pr=6174#step:5:16)7,32): error TS2362: The left-hand side of an arithmetic operation must be of type 'any', 'number', 'bigint' or an enum type.
Error: src/devices/tuya.ts(6[16](https://github.com/Koenkk/zigbee-herdsman-converters/actions/runs/6239136324/job/16936417150?pr=6174#step:5:17)1,25): error TS2365: Operator '<' cannot be applied to types 'unknown' and 'number'.
Error: src/devices/tuya.ts(6162,32): error TS2365: Operator '+' cannot be applied to types '4096' and 'unknown'.
Error: Process completed with exit code 2.
``` 
What can I change? Should "value" be declared like this?
```javascript
            [27,  'local_temperature_calibration',  {
                from: (v: Buffer) => {
                    if (v > 4000) 
                        return v - 4096;
                    return v;
                    },
                to: (v) => {
                    if (v < 0) 
                        return 4096 + v;
                    return v;
                    }
                }],
```

Reply:
Updated your code, please do a final check and then I can merge it.

Reply:
Checked!
Please merge it.